pkg/cache/external/scheduler_cache.go

```diff
@@ -36,23 +36,23 @@ import (
 // nodes are cached in the form of de-scheduler nodeInfo, instead of re-creating all nodes info from scratch,
 // we replicate nodes info from de-scheduler, in order to re-use predicates functions.
 type SchedulerCache struct {
-	// node name to NodeInfo map
-	nodesMap map[string]*framework.NodeInfo
-	podsMap  map[string]*v1.Pod
-	// this is a map of assumed pods,
-	// the value indicates if a pod volumes are all bound
-	assumedPods map[string]bool
-	lock        sync.RWMutex
-	// client APIs
-	clients *client.Clients
+	nodesMap              map[string]*framework.NodeInfo // node name to NodeInfo map
+	podsMap               map[string]*v1.Pod
+	assumedPods           map[string]bool   // map of assumed pods, value indicates if pod volumes are all bound
+	pendingAllocations    map[string]string // map of pod to node ID, presence indicates a pending allocation for scheduler
+	inProgressAllocations map[string]string // map of pod to node ID, presence indicates an in-process allocation for scheduler
```

Comment:
We need some extra comment which explains what is the difference between a "pending" and an "in progress" allocation. Just by looking at the code, it's not straightforward which comes first and what is the lifecycle of the pods.

Reply:
A pending allocation is one which YuniKorn has decided on, but which has not yet been communicated to the default scheduler via the scheduling framework (the PreFilter()/Filter() calls in the plugin). An in-progress allocation is one which HAS been communicated to the default scheduler via PreFilter() / Filter() but which we have not yet got confirmation of the Pod being bound (PostBind() in the plugin has not yet been called). This is used to track state to ensure that we don't reprocess the same Pod or leak in the case where the default scheduler is unable to bind the pod for some reason.

Reply:
Added additional documentation in scheduler_cache.go.

Reply:
Thanks, this definitely helps to understand the code.